test/integration/policy_set_test.go

```diff
@@ -0,0 +1,247 @@
+	"github.com/stolostron/governance-policy-propagator/controllers/common"
+	"github.com/stolostron/governance-policy-propagator/test/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	testcommon "github.com/stolostron/governance-policy-framework/test/common"
+)
+
+const (
+	testPolicyName          string = "test-policy"
+	testPolicySetName       string = "test-policyset"
+	testPolicyYaml          string = "../resources/policy_set/test-policy.yaml"
+	testPolicySetPatchYaml  string = "../resources/policy_set/patch-policy-set.yaml"
+	testedDisablePolicyYaml string = "../resources/policy_set/disable-policy.yaml"
+)
+
+var _ = Describe("GRC: [P1][Sev1][policy-grc] Test policy set", func() {
+	Describe("Create policy, policyset, and placement in ns:"+userNamespace, func() {
+		It("Should create and process policy and policyset", func() {
+			By("Creating " + testPolicyYaml)
+			output, err := testcommon.OcHub("apply",
+				"-f", testPolicyYaml,
+				"-n", userNamespace)
+			By("Creating " + testPolicyYaml + " result is " + output)
+			Expect(err).To(BeNil())
+
+			By("Checking that the root policy was created")
+			rootPolicyRsrc := clientHubDynamic.Resource(testcommon.GvrPolicy)
+			var rootPolicy *unstructured.Unstructured
+			Eventually(
+				func() error {
+					var err error
+					rootPolicy, err = rootPolicyRsrc.Namespace(userNamespace).Get(
+						context.TODO(), testPolicyName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+
+			templates, found, err := unstructured.NestedSlice(rootPolicy.Object, "spec", "policy-templates")
+			Expect(err).Should(BeNil())
+			Expect(found).Should(BeTrue())
+			Expect(len(templates)).Should(Equal(1))
+
+			By("Patching placement rule " + testPolicyName + "-plr")
+			output, err = utils.KubectlWithOutput("patch", "-n", userNamespace, "placementrule.apps.open-cluster-management.io/"+testPolicySetName+"-plr",
+				"--type=json", "-p=[{\"op\": \"replace\", \"path\": \"/spec/clusterSelector/matchExpressions\", \"value\":[{\"key\": \"name\", \"operator\": \"In\", \"values\": ["+clusterNamespace+"]}]}]",
+				"--kubeconfig="+kubeconfigHub)
+			By("Patching placement rule result is " + output)
+			Expect(err).To(BeNil())
+
+			By("Checking " + testPolicyName + " on managed cluster in ns " + clusterNamespace)
+			policyRsrc := clientHubDynamic.Resource(testcommon.GvrPolicy)
+			Eventually(
+				func() error {
+					var err error
+					_, err = policyRsrc.Namespace(clusterNamespace).Get(
+						context.TODO(), userNamespace+"."+testPolicyName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+
+			opt := metav1.ListOptions{
+				LabelSelector: common.RootPolicyLabel + "=" + userNamespace + "." + testPolicyName,
+			}
+
+			By("Patching replicated policy statuses - NonCompliant")
```

Comment:
Patching the status of replicated policy is not needed. This is running against live clusters so the status will be generated by controller.

Reply:
Should just check the status of replicated policy